AutomatticTracks/src/test/java/com/automattic/android/tracks/CrashLogging/CrashLoggingTest.kt

```diff
@@ -0,0 +1,70 @@
+package com.automattic.android.tracks.crashlogging
+
+import com.automattic.android.tracks.TracksUser
+import com.automattic.android.tracks.fakes.FakeDataProvider
+import com.automattic.android.tracks.fakes.testUser
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.never
+import org.mockito.Mockito.spy
+import org.mockito.Mockito.times
+import org.mockito.Mockito.verify
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+
+@RunWith(RobolectricTestRunner::class)
+class CrashLoggingTest {
+
+    lateinit var dataProvider: CrashLoggingDataProvider
+    var user: TracksUser? = null
+
+    private fun initialize(currentUser: TracksUser? = null) {
+
+        user = if (currentUser != null) spy(currentUser) else null
+
+        dataProvider = spy(FakeDataProvider(currentUser = user))
+
+        CrashLogging.start(
+            context = RuntimeEnvironment.systemContext,
+            dataProvider = dataProvider
+        )
+    }
+
+    @Test(expected = UnsupportedOperationException::class)
+    fun shouldThrowExceptionOnRequest() {
+        CrashLogging.crash()
+    }
```

Comment:
Hmm.. I am a little confused. This seems to just call a function that throws an exception. The `OnRequest` part doesn't seem to come into play at all, as in, we don't seem to be testing anything in the `CrashLogging` class. Am I missing something?

Reply:
> we don't seem to be testing anything in the CrashLogging class

I think we do test something, we test that `CrashLogging#crash` method will throw an exception, which is expected behavior. Naming it `OnRequest` was about `request` from client/library consumer in a sense of "when client wants to". 

Eventually, as discussed here https://github.com/Automattic/Automattic-Tracks-Android/pull/51#discussion_r599870154 I'll remove `CrashLogging#crash` so this test won't cause any confusion anymore.